Lib/pathlib.py

```diff
@@ -195,10 +195,9 @@ def _select_unique(paths):
     yielded = set()
     try:
         for path in paths:
-            raw_path = path._raw_path
-            if raw_path not in yielded:
+            if path not in yielded:
                 yield path
-                yielded.add(raw_path)
```

Comment:
This is slightly slower, but it shouldn't matter. To hit this code path, users need to supply a pattern with two non-consecutive `**` components, such as `**/*/**`, and leave the _follow_symlinks_ argument with its default value of `None` (rather than setting it to `True` or `False`).